StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1009UnitTests.cs

```diff
@@ -0,0 +1,557 @@
+        }
+
+        [Fact]
+        public async Task TestAttributeWithParametersAndNoSpaceAfterClosingParenthesisAsync()
+        {
+            const string testCode = @"using System;
+using System.Security.Permissions;
+
+[PermissionSet(SecurityAction.LinkDemand, Name = ""FullTrust"")]
+public class Foo
+{
+    public void Method(int param1, int param2)
+    {
+    }
+}";
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestAttributeWithParametersAndSpaceAfterClosingParenthesisAsync()
+        {
+            const string testCode = @"using System;
+using System.Security.Permissions;
+
+[PermissionSet(SecurityAction.LinkDemand, Name = ""FullTrust"") ]
+public class Foo
+{
+    public void Method(int param1, int param2)
+    {
+    }
+}";
+            DiagnosticResult expected = this.CSharpDiagnostic().WithArguments(" not", "followed").WithLocation(4, 61);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestConstructorInheritenceWithSpaceAfterClosingParenthesisAsync()
+        {
+            const string testCode = @"using System;
+
+public class Foo
+{
+    public Foo(int i) : base()
+    {
+    }
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestConstructorInheritenceWithNoSpaceAfterClosingParenthesisAsync()
+        {
+            const string testCode = @"using System;
+
+public class Foo
+{
+    public Foo(int i): base()
+    {
+    }
+}";
+            DiagnosticResult expected = this.CSharpDiagnostic().WithArguments(string.Empty, "followed").WithLocation(5, 21);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestLambdaExpressionWithSpaceAfterClosingParenthesisAsync()
+        {
+            var validStatement = @"System.EventHandler handler = (s, e) => { };";
+
+            await this.TestWhitespaceInStatementOrDeclAsync(validStatement, EmptyDiagnosticResults).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestLambdaExpressionWithNoSpaceAfterClosingParenthesisAsync()
+        {
+            var invalidStatement = @"System.EventHandler handler = (s, e)=> { };";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithArguments(string.Empty, "followed").WithLocation(7, 48);
+
+            await this.TestWhitespaceInStatementOrDeclAsync(invalidStatement, expected).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSpaceAfterParenthisisInAddOperationAsync()
```

Comment:
You can instead of using Fact do something like this:
```csharp
[Theory]
[InlineData("+")]
[InlineData("-")]
[InlineData("*")]
[InlineData("/")]
TestSpaceAfterParenthisisInOperationAsync(string operator) { ... }
```
xunit will call this method for each of these operators and you have to write this method only once.

NIT means nitpicking. So not very important